docs/quickstart.rst

```diff
@@ -8,51 +8,140 @@ up pg_auto_failover to replicate data between them. We’ll simulate failure in
 the primary node and see how the system smoothly switches (fails over)
 to the secondary.
 
-For simplicity we’ll run all the pieces on a single machine, but in
-production there would be three independent machines involved for each
-managed PostgreSQL service. One machine for the primary, another for
-the secondary, and the last as a monitor which watches the other nodes’
-health, manages global state, and assigns nodes their roles.
+For illustration, we'll run our databases on virtual machines in the Azure
+platform, but the techniques here are relevant to any cloud provider or
+on-premise network. We'll use three virtual machines: a primary database, a
+secondary database, and a monitor which watches the other nodes’ health,
+manages global state, and assigns nodes their roles.
+
+.. _quickstart_network:
+
+Create virtual network
+----------------------
+
+Our database machines need to talk to each other and to the monitor node, so
+let's create a virtual network.
+
+.. code-block:: bash
+
+   az group create \
+       --name ha-demo \
+       --location eastus
+
+   az network vnet create \
+       --resource-group ha-demo \
+       --name ha-demo-net \
+       --address-prefix 10.0.0.0/16
+
+We need to open ports 5432 (Postgres) and 22 (SSH) between the machines, and
+also give ourselves address from our remote IP. We'll do this with a network
+security group and a subnet.
+
+.. code-block:: bash
+
+   az network nsg create \
+       --resource-group ha-demo \
+       --name ha-demo-nsg
+
+   export MY_IP=`dig +short myip.opendns.com @resolver1.opendns.com`
+
+   az network nsg rule create \
+       --resource-group ha-demo \
+       --nsg-name ha-demo-nsg \
+       --name ha-demo-ssh-and-pg \
+       --access allow \
+       --protocol Tcp \
+       --direction Inbound \
+       --priority 100 \
+       --source-address-prefixes $MY_IP 10.0.1.0/24 \
+       --source-port-range "*" \
+       --destination-address-prefix "*" \
+       --destination-port-ranges 22 5432
+
+   az network vnet subnet create \
+       --resource-group ha-demo \
+       --vnet-name ha-demo-net \
+       --name ha-demo-subnet \
+       --address-prefixes 10.0.1.0/24 \
+       --network-security-group ha-demo-nsg
+
+Finally add three virtual machines (ha-demo-a, ha-demo-b, and
+ha-demo-monitor).
+
+.. code-block:: bash
+
+   for node in monitor a b app
+   do
+   az vm create \
+       --resource-group ha-demo \
+       --name ha-demo-${node} \
+       --vnet-name ha-demo-net \
+       --subnet ha-demo-subnet \
+       --nsg ha-demo-nsg \
+       --public-ip-address ha-demo-${node}-ip \
+       --image UbuntuLTS \
+       --admin-username ha-admin \
+       --generate-ssh-keys
+   done
+
+List and record the public IP addresses that this command outputs for each
+machine:
+
+.. code-block:: bash
+
+   az vm list-ip-addresses --ids \
+       $(az vm list -g ha-demo --query "[].id" -o tsv)
```

Comment:
Check that one out! 

```
az vm list-ip-addresses --ids  $(az vm list -g ha-demo --query "[].id" -o tsv) --query '[] [] .virtualMachine .{Name: name, PrivateIP: network.privateIpAddresses[0], PublicIP: network.publicIpAddresses[0].ipAddress}' -o table
```

Reply:
I liked this better

Reply:
I found how to sort the array, here by virtual machine name. It's possible to sort by privateIp but then `.10` is listed before `.1` because of alphabetic rules, so that's not as useful. I didn't play with trying to map the IP string into a number for the sorting expression...

```
az vm list-ip-addresses \
  --ids  $(az vm list -g ha-demo --query "[].id" -o tsv) \
  -o table \
  --query 'sort_by( [] [] 
             .virtualMachine 
             .{Name: name,
               PrivateIP: network.privateIpAddresses[0], 
               PublicIP: network.publicIpAddresses[0].ipAddress
             }, &Name)'

Name             PrivateIP    PublicIP
---------------  -----------  -------------
ha-demo-a        10.0.1.5     13.92.189.9
ha-demo-app      10.0.1.7     40.80.147.143
ha-demo-b        10.0.1.6     40.71.100.103
ha-demo-c0       10.0.1.8     13.90.42.3
ha-demo-c1       10.0.1.9     13.90.41.111
ha-demo-monitor  10.0.1.4     13.92.243.131
ha-demo-w1       10.0.1.10    13.90.40.182
ha-demo-w2       10.0.1.11    13.90.41.200
ha-demo-w3       10.0.1.12    13.68.225.239
ha-demo-w4       10.0.1.13    13.92.80.219
```